### ISSUE TITLE
Error using lets-plot in streamlit 

### ISSUE THREAD

**Issue by schakalakka** [JetBrains/lets-plot#595]
I tried using lets-plot in streamlit through the streamlit-letsplot package. 

But I am getting the following error: 
```
generateStaticHtmlPage() - Exception: Trying to access top level value not marked as @ThreadLocal or @SharedImmutable from non-main thread
```
I already mentioned on the Github page of [streamlit-letsplot](https://github.com/randyzwitch/streamlit-letsplot/issues/8) but I think it has something to do with the underlying Kotlin code, right? 

Example code 
```python 
np.random.seed(12)
data = dict(
    cond=np.repeat(['A','B'], 200),
    rating=np.concatenate((np.random.normal(0, 1, 200), np.random.normal(1, 1.5, 200)))
)

a = ggplot(data, aes(x='rating', fill='cond')) + ggsize(500, 250) \
+ geom_density(color='dark_green', alpha=.7) + scale_fill_brewer(type='seq') \
+ theme(axis_line_y='blank')

st_letsplot(a)

**Comment by IKupriyanov-HORIS** [comment]
Hello! Yeah, it's a K/N memory model feature.   
Not sure there are no more such issues (no linting), but demo works fine:
![image](https://user-images.githubusercontent.com/14200189/189910296-19fa8031-7baa-4ae8-81f4-8fbb65807179.png)

But there is a good news - the problem will be totally eliminated in Kotlin 1.7.20 (https://youtrack.jetbrains.com/issue/KT-36799)

**Comment by schakalakka** [comment]
Does the new Kotlin version automatically ship with the lets-plot package (for python) or how does it work?

**Comment by alshan** [comment]
Yes. As soon as we switch our build scripts to Kotlin 1.7, the Lets-Plot binaries (the Python package) will be free of issues fixed in Kotlin 1.7.
But even if we do not switch to Kotlin 1.7 , this fix by @IKupriyanov-HORIS  may help you. Let's wait for the next release of lets-plot.
